### PR TITLE
fix: correct failing power limit tests

### PR DESCRIPTION
**Pull request by shubskmr**

### CODE REVIEW

**Review by narenbagria — APPROVED** [review]

File: conformance_tests/sysman/test_sysman_power/src/test_sysman_power.cpp [hunk 1]
```diff
@@ -490,35 +490,39 @@ TEST_F(
...
-        zes_power_limit_ext_desc_t power_sustained_get = {};
-
-        auto power_limits_descriptors_get =
-            lzt::get_power_limits_ext(p_power_handle, &count_power);
-        for (auto p_power_limits_descriptor_get :
-             power_limits_descriptors_get) {
-          if (p_power_limits_descriptor_get.level ==
-              ZES_POWER_LEVEL_SUSTAINED) {
-            power_sustained_get = p_power_limits_descriptor_get;
+      if (power_sustained_set.level == ZES_POWER_LEVEL_SUSTAINED) {
```
Comment (jitendrasharma1989) [inline conformance_tests/sysman/test_sysman_power/src/test_sysman_power.cpp:493]: Could you please share what this change is fixing? Could you attach a related to jira?

Comment (shubskmr) [reply conformance_tests/sysman/test_sysman_power/src/test_sysman_power.cpp:493]: In this test a empty structure is 0 initialized.
`zes_power_limit_ext_desc_t power_sustained_set = {};`

It then iterates over all power limit descriptors and tries to find the ZES_POWER_LEVEL_SUSTAINED limit descriptor to fill `power_sustained_set ` with.

The problem here was that there are some platforms where  ZES_POWER_LEVEL_SUSTAINED will not be available and `power_sustained_set ` will be empty. Hence we should first check if it really has data for ZES_POWER_LEVEL_SUSTAINED before proceeding with rest of the test.

There is no Jira for this as I just observed them failing on elmo for long time and fixed them.

Comment (jitendrasharma1989) [reply conformance_tests/sysman/test_sysman_power/src/test_sysman_power.cpp:493]: So how could we validate condition where on supported platforms ZES_POWER_LEVEL_SUSTAINED will not be available?

Comment (shubskmr) [reply conformance_tests/sysman/test_sysman_power/src/test_sysman_power.cpp:493]: I don't think there is a way to know available limits before hand and configure CTS accordingly for different platforms.

Comment (jitendrasharma1989) [reply conformance_tests/sysman/test_sysman_power/src/test_sysman_power.cpp:493]: @AshwinKumarKulkarni @saik-intel @vishnu-khanth I feel this validation could be done in embargo tests.

File: conformance_tests/sysman/test_sysman_power/src/test_sysman_power.cpp [hunk 2]
```diff
@@ -490,35 +490,39 @@ TEST_F(
...
+          lzt::set_power_limits_ext(
+              p_power_handle, &count_power,
+              power_limits_descriptors.data()); // set power limits for all
+                                                // descriptors
+
+          zes_power_limit_ext_desc_t power_sustained_get = {};
+
+          auto power_limits_descriptors_get =
+              lzt::get_power_limits_ext(p_power_handle, &count_power);
+          for (auto p_power_limits_descriptor_get :
```
Comment (jitendrasharma1989) [inline conformance_tests/sysman/test_sysman_power/src/test_sysman_power.cpp:504]: minor: could we use const auto& in this for loop and all others. Other than that comment, PR looks fine.

File: conformance_tests/sysman/test_sysman_power/src/test_sysman_power.cpp [hunk 3]
```diff
@@ -852,34 +856,37 @@ TEST_F(
...
+          EXPECT_LE(power_sustained_get.limit, power_sustained_getMax.limit);
+        } else {
+          LOG_INFO << "Set limit not supported due to sustained "
+                      "limitValueLocked flag is true";
         }
-        EXPECT_LE(power_sustained_get.limit, power_sustained_getMax.limit);
       } else {
-        LOG_INFO << "Set limit not supported due to sustained "
-                    "limitValueLocked flag is true";
+          LOG_INFO << "Sustained power limit not supported";
```
Comment (vishnu-khanth) [inline conformance_tests/sysman/test_sysman_power/src/test_sysman_power.cpp:889]: I think this info can be outside for loop, after iterating over all descriptors

File: conformance_tests/sysman/test_sysman_power/src/test_sysman_power.cpp [hunk 4]
```diff
@@ -894,44 +901,50 @@ TEST_F(
...
+    uint32_t single_count = 1;
+    
     auto power_limits_descriptors =
         lzt::get_power_limits_ext(power_card_handle, &count_power);
     for (int i = 0; i < power_limits_descriptors.size(); i++) {
+      zes_power_limit_ext_desc_t power_peak_initial = {};
+      zes_power_limit_ext_desc_t power_peak_Max = {};
+      zes_power_limit_ext_desc_t power_peak_getMax = {};
+      zes_power_limit_ext_desc_t power_peak_get = {};
       if (power_limits_descriptors[i].level == ZES_POWER_LEVEL_PEAK) {
```
Comment (vishnu-khanth) [inline conformance_tests/sysman/test_sysman_power/src/test_sysman_power.cpp:913]: Similarly info log can be added if Peak level is not available after iterating over all descriptors

File: conformance_tests/sysman/test_sysman_power/src/test_sysman_power.cpp (hunk 3 above)
Comment (shubskmr) [reply conformance_tests/sysman/test_sysman_power/src/test_sysman_power.cpp]: done

File: conformance_tests/sysman/test_sysman_power/src/test_sysman_power.cpp (hunk 4 above)
Comment (shubskmr) [reply conformance_tests/sysman/test_sysman_power/src/test_sysman_power.cpp:917]: done

File: conformance_tests/sysman/test_sysman_power/src/test_sysman_power.cpp (hunk 2 above)
Comment (shubskmr) [reply conformance_tests/sysman/test_sysman_power/src/test_sysman_power.cpp]: done

**Review by jitendrasharma1989 — APPROVED** [review]

**Review by vishnu-khanth — APPROVED** [review]

**Review by jitendrasharma1989 — APPROVED** [review]

**Review by AshwinKumarKulkarni — APPROVED** [review]

**Pull request merged**